dashboard/test/controllers/courses_controller_test.rb

```diff
@@ -94,14 +94,14 @@ class CachedQueryCounts < ActionController::TestCase
 
     test 'student views course overview with caching enabled' do
       sign_in create(:student)
-      assert_cached_queries(5) do
+      assert_cached_queries(6) do
```

Comment:
just removing the old ability check increased this query count and the one below by one, so there must have been something later on that was able to reuse the teachers query in that check.